pdfarranger: Add macOS support

Endle opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
pdfarranger has macOS support in 1.11.1:  https://github.com/pdfarranger/pdfarranger/commit/5e4db7cddccecdfe034841512cb9860892473921

At nixpkgs side, we just need to set up the lib path.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Endle committed:
pdfarranger: Add macOS support

Endle committed:
Fix code format

symphorien commented on pkgs/by-name/pd/pdfarranger/package.nix:
```diff
@@ -6,6 +6,7 @@
   gtk3,
   poppler_gi,
   libhandy,
```
please take stdenv, gettext as argument instead of pkgs. I think pkgs breaks splicing, which is important for cross compilation.

symphorien commented on pkgs/by-name/pd/pdfarranger/package.nix:
```diff
@@ -20,7 +21,18 @@ python3Packages.buildPythonApplication rec {
+    ''''
+    + lib.optionals pkgs.stdenv.isDarwin ''
+      LINTL="${lib.getLib pkgs.gettext}/lib/libintl.8.dylib"
+      sed -i "s@return 'libintl.8.dylib'@return '$LINTL'@" pdfarranger/pdfarranger.py
```
please use `substituteInPlace --replace-fail` or a patch instead. If the code you are patching changes, sed will silently do nothing and linux people will not notice the breakage.

symphorien commented on pkgs/by-name/pd/pdfarranger/package.nix:
```diff
@@ -20,7 +21,18 @@ python3Packages.buildPythonApplication rec {
+      pkgs.gettext
+    ];
+
+  patchPhase =
+    ''''
+    + lib.optionals pkgs.stdenv.isDarwin ''
```
```suggestion
  postPatch = lib.optionals pkgs.stdenv.isDarwin ''
```

this way adding `patches =` (for example in an override) still works.

symphorien replied:
actually this should use optionalString, not optionals. I wonder how this slipped through nix-review...

symphorien reviewed (commented):
Thank you for your PR. As I can't maintain pdfarranger on mac, would you mind adding yourself as maintainer?

Endle committed:
patch improvement

1. add myself as maintainer
2. Use stdenv and gettext, remove pkgs
3. Use substituteInPlace
4. Use postPatch
5. Use optionalString

Endle committed:
Run nixfmt

Endle commented:
@symphorien Thank you so much for your suggestions!

My updated version has such changes:

    1. add myself as maintainer
    2. Use stdenv and gettext, remove pkgs
    3. Use substituteInPlace
    4. Use postPatch
    5. Use optionalString
    6. nixfmt

symphorien commented:
Thank you!

Note for next time: please squash the changes you perform as part of the review process. See https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions for more details

Endle commented:
Thank you!

> Note for next time: please squash the changes you perform as part of the review process.

Thanks for the reminder. Next time I'll squash my patches at my side
